modules/phase_field/src/kernels/PFFractureBulkRate.C

```diff
@@ -0,0 +1,105 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+#include "PFFractureBulkRate.h"
+#include "MooseVariable.h"
+#include "RankTwoTensor.h"
+#include "MathUtils.h"
+
+template <>
+InputParameters
+validParams<PFFractureBulkRate>()
+{
+  InputParameters params = validParams<PFFractureBulkRateBase>();
+  params.addClassDescription(
+      "Kernel to compute bulk energy contribution to damage order parameter residual equation");
+  return params;
+}
+
+PFFractureBulkRate::PFFractureBulkRate(const InputParameters & parameters)
+  : PFFractureBulkRateBase(parameters), _second_u(second())
+{
+}
+
+Real
+PFFractureBulkRate::computeQpResidual()
+{
+  const Real & gc = _gc_prop[_qp];
+  const Real & c = _u[_qp];
+  const Real beta = _second_u[_qp].tr();
+  const Real x2 = 2.0 * (1.0 - c) * _G0_pos[_qp] / gc - c / _width;
+  const Real x1 = _width * beta;
+
+  /**
+   * This if implements the <...>+ positive part operator. To determine if the
+   * expression is positive we use the original form that is _not_ integrated by parts.
+   * The residual we return, however, replaces x1 with the expression that is integrated by parts
+   * once (_second_u[_qp].tr() * _test[_i][_qp] -> -_grad_u[_qp] * _grad_test[_i][_qp]).
+   * This allows the user to use C1 shape functions (such as second order Lagrange).
```

Comment:
The definition of C^m continuity is that the function's derivative of order m is continuous. So all the Lagrange shape functions are C^0 and Hermite shape functions are C^1. :-)